### ISSUE TITLE
Prevent Segmentation Faults when adding arcs before adding node.

### ISSUE THREAD

**Issue by Agrover112** [gtn-org/gtn#26]
Idk if this should be fixed, but could this be handled with an exception? or ..?]
![image](https://user-images.githubusercontent.com/42321810/152578821-8bdb46bf-279e-4d18-a5db-f6540cbc5019.png)

**Comment by awni** [comment]
Yes, I will look into adding a test and an exception for this if it doesn't slow down the graph building process too much.

**Comment by Agrover112** [comment]
> Yes, I will look into adding a test and an exception for this if it doesn't slow down the graph building process too much.

Cool, let me know if I could help with this if at all, although I can't promise  i am comfortable with C++ tbh.
